### PR TITLE
fix: status item styles, add option to place status icon before status text

### PR DESCRIPTION
**Pull request by fmehta-eightfold**
## SUMMARY:

- **Feature: Option to place status text after the status icon**
- **Fix status item text font size** - according to the design, the font should be `2px` larger than the icon, and have `font-weight: 600`

## JIRA TASK (Eightfold Employees Only):
ENG-40399

## CHANGE TYPE:

- [x] Bugfix Pull Request
- [x] Feature Pull Request

## TEST COVERAGE:

- [ ] Tests for this change already exist
- [ ] I have added unittests for this change

## TEST PLAN:
Story book for Avatar should get updated to reflect the above mentioned feature and fix

**Before**
image

**After**
image

### CODE REVIEW

File: src/components/Avatar/Avatar.tsx [hunk 1]
```diff
@@ -200,21 +220,20 @@ const AvatarStatusItems: FC<BaseAvatarProps> = React.forwardRef(
...
-                  }}
-                  className={styles.avatarStatusItemText}
-                >
-                  {statusItemProps.text}
-                </span>
-              ) : (
-                ''
-              )}
-              <Icon {...statusItemProps} />
+              {showStatusItemText &&
```
Comment (dkilgore-eightfold) [inline src/components/Avatar/Avatar.tsx:223]: No need for a ternary here, use `&&`.

File: src/components/Avatar/Avatar.tsx [hunk 2]
```diff
@@ -200,21 +220,20 @@ const AvatarStatusItems: FC<BaseAvatarProps> = React.forwardRef(
...
+              (showStatusItemsText[position] ?? true) &&
+              !statusItemProps.placeTextAfterIcon
+                ? statusItemTextElement
+                : ''}
+              <Icon
+                {...statusItemProps}
+                classNames={styles.avatarStatusItemIcon}
+              />
+              {showStatusItemText &&
```
Comment (dkilgore-eightfold) [inline src/components/Avatar/Avatar.tsx:232]: Same as above.

File: src/components/Avatar/Avatar.types.ts [hunk 3]
```diff
@@ -67,6 +67,11 @@ export interface StatusItemsProps extends IconProps {
    * Class for status item wrapper
    */
   wrapperClassName?: string;
+  /**
+   * Place text after the icon if true
+   * @default false
+   */
+  placeTextAfterIcon?: boolean;
```
Comment (dkilgore-eightfold) [inline src/components/Avatar/Avatar.types.ts:74]: In other components, the pattern is `alignIcon` and the text position is persistent.

**Review by dkilgore-eightfold — CHANGES_REQUESTED** [review]

File: src/components/Avatar/Avatar.tsx (hunk 1 above)
Comment (fmehta-eightfold) [reply src/components/Avatar/Avatar.tsx]: Done

File: src/components/Avatar/Avatar.tsx (hunk 2 above)
Comment (fmehta-eightfold) [reply src/components/Avatar/Avatar.tsx]: Done

File: src/components/Avatar/Avatar.types.ts (hunk 3 above)
Comment (fmehta-eightfold) [reply src/components/Avatar/Avatar.types.ts]: Updated

**Review by dkilgore-eightfold — APPROVED** [review]

**Comment by dkilgore-eightfold** [comment]
@fmehta-eightfold Fix the stories file with `alignIcon` and you should be good to go.

**Pull request merged**